src/org/anjocaido/groupmanager/dataholder/WorldDataHolder.java

```diff
@@ -481,11 +478,9 @@ public boolean haveGroupsChanged() {
 		if (groups.isGroupsChanged()) {
 			return true;
 		}
-		synchronized (groups) {
-			for (Group g : groups.getGroups().values()) {
-				if (g.isChanged()) {
-					return true;
-				}

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build